packages/babel-helpers/src/helpers.js

```diff
@@ -861,8 +861,13 @@ helpers.arrayWithHoles = helper("7.0.0-beta.0")`
 helpers.iterableToArray = helper("7.0.0-beta.0")`
   export default function _iterableToArray(iter) {
     if (
-      Symbol.iterator in Object(iter) ||
-      Object.prototype.toString.call(iter) === "[object Arguments]"
+      Array.isArray(iter)
+      || typeof iter === 'string'
+      || (typeof Symbol === 'function' && Symbol.iterator in Object(iter))
+      || (iter && 'length' in iter)
+      || Object.prototype.toString.call(iter) === "[object Map]"
+      || Object.prototype.toString.call(iter) === "[object Set]"
+      || Object.prototype.toString.call(iter) === "[object Arguments]"
```

Comment:
Calling `Object#toString` 3 times does not look like something good. It would be better to cache the result.

Reply:
In the common case, it will shortcircuit after the Symbol.iterator check, at least